packages/bundler-sdk-viem/src/ActionBundle.ts

```diff
@@ -0,0 +1,45 @@
+import type { SimulationResult } from "@morpho-org/simulation-sdk";
+import type { Account, Address, Chain, Client, Hex, Transport } from "viem";
+import { BundlerAction } from "./BundlerAction.js";
+import type {
+  Action,
+  SignatureRequirement,
+  TransactionRequirement,
+} from "./types/index.js";
+
+export class ActionBundleRequirements<
+  TR extends { tx: { to: Address; data: Hex } } = TransactionRequirement,
+  SR extends SignatureRequirement = SignatureRequirement,
+> {
+  constructor(
+    public readonly txs: TR[] = [],
+    public readonly signatures: SR[] = [],
+  ) {}
+
+  sign(client: Client<Transport, Chain | undefined, Account>): Promise<Hex[]>;
+  sign(client: Client, account: Account): Promise<Hex[]>;
+  sign(client: Client, account: Account = client.account!) {
+    return Promise.all(
+      this.signatures.map((requirement) => requirement.sign(client, account)),
+    );
+  }
+}
```

Comment:
This doesn't work with a browser wallet, requirements need to be signed one by one